weave-js/src/components/PagePanelComponents/Home/Browse3/filters/SelectDatetimeDropdown.tsx

```diff
@@ -0,0 +1,379 @@
+  isHovered: boolean;
+  onClick: () => void;
+  onMouseEnter: () => void;
+  onMouseLeave: () => void;
+};
+
+const SuggestionItem: React.FC<SuggestionItemProps> = ({
+  suggestion,
+  index,
+  isSelected,
+  isHovered,
+  onClick,
+  onMouseEnter,
+  onMouseLeave,
+}) => {
+  return (
+    <li
+      key={index}
+      role="option"
+      aria-selected={isSelected}
+      onClick={onClick}
+      onMouseEnter={onMouseEnter}
+      onMouseLeave={onMouseLeave}
+      style={{
+        padding: '8px',
+        cursor: 'pointer',
+        backgroundColor: isSelected
+          ? isHovered
+            ? MOON_100
+            : '#E1F7FA' // special super light teal
+          : isHovered
+          ? MOON_100
+          : WHITE,
+      }}
+      onMouseDown={e => e.preventDefault()}>
+      <div
+        style={{
+          display: 'flex',
+          justifyContent: 'space-between',
+          alignItems: 'center',
+        }}>
+        <span>{suggestion.abbreviation}</span>
+        <span style={{color: MOON_500}}>{suggestion.label}</span>
+      </div>
+    </li>
+  );
+};
+
+type SuggestionsListProps = {
+  isDropdownVisible: boolean;
+  dropdownRef: React.RefObject<HTMLUListElement>;
+  predefinedSuggestions: PredefinedSuggestion[];
+  selectedSuggestion: string | null;
+  hoveredIndex: number | null;
+  handleSuggestionClick: (suggestionValue: string) => void;
+  handleMouseEnter: (index: number) => void;
+  handleMouseLeave: () => void;
+};
+
+const SuggestionsList: React.FC<SuggestionsListProps> = ({
+  isDropdownVisible,
+  dropdownRef,
+  predefinedSuggestions,
+  selectedSuggestion,
+  hoveredIndex,
+  handleSuggestionClick,
+  handleMouseEnter,
+  handleMouseLeave,
+}) => {
+  if (!isDropdownVisible) {
+    return null;
+  }
+
+  return (
+    <ul
+      ref={dropdownRef}
+      role="listbox"
+      style={{
+        position: 'absolute',
+        top: '100%',
+        left: '0',
+        right: '0',
+        backgroundColor: WHITE,
+        border: `1px solid ${MOON_200}`,
+        borderRadius: '4px',
+        marginTop: '4px',
+        listStyle: 'none',
+        fontSize: '14px',
+        padding: '0',
+        overflow: 'hidden',
+        zIndex: 1000,
+      }}>
+      {predefinedSuggestions.map((suggestion, index) => (
+        <SuggestionItem
+          key={index}
+          suggestion={suggestion}
+          index={index}
+          isSelected={selectedSuggestion === suggestion.abbreviation}
+          isHovered={hoveredIndex === index}
+          onClick={() => handleSuggestionClick(suggestion.abbreviation)}
+          onMouseEnter={() => handleMouseEnter(index)}
+          onMouseLeave={handleMouseLeave}
+        />
+      ))}
+    </ul>
+  );
+};
+
+/**
+ * Fires an analytics event when the metrics plots are viewed.
+ * This is used to track the usage and latency of the metrics plots.
+ * Only fires once when opened.
+ */
+const useFireAnalyticsForDateFilterDropdownUsed = (
+  entity: string,
+  project: string,
```

Comment:
We are drilling entity and project through a bunch of components now just to include them in analytics. If we could pick entity and project up from an existing hook it would make the diff smaller and the code cleaner.

Reply:
yeah... but unfortunately in this case we can't. https://weightsandbiases.slack.com/archives/C03BSTEBD7F/p1742413436662399